src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -6008,20 +6034,30 @@ void func_8083C148(Player* this, PlayState* play) {
     this->stateFlags1 &= ~(PLAYER_STATE1_13 | PLAYER_STATE1_14 | PLAYER_STATE1_20);
 }
 
-s32 Player_ActionChange_6(Player* this, PlayState* play) {
+/**
+ * Handles some specific behavior related to the A button.
+ *
+ * First, if a roll can be performed, that will take precedence.
+ * The next option is putting away an item in hand, if applicable.
+ * Lastly if neither of those two options are available, Navi can be toggled on and off.
+ * She will either appear and fly around Link's head, or fly back into his body.
+ *
+ * Note that the item putaway and Navi cases do not trigger new actions. In these cases, `false`
+ * is returned and the action processing the Action Change List will remain the same.
+ */
+s32 Player_ActionChange_HandleRollingAndPutAway(Player* this, PlayState* play) {
     if (!func_80833B54(this) && !sUpperBodyIsBusy && !(this->stateFlags1 & PLAYER_STATE1_23) &&
         CHECK_BTN_ALL(sControlInput->press.button, BTN_A)) {
-        if (func_8083BC7C(this, play)) {
-            return 1;
-        }
-        if ((this->unk_837 == 0) && (this->heldItemAction >= PLAYER_IA_SWORD_MASTER)) {
+        if (Player_TryRolling(this, play)) {
+            return true;
+        } else if ((this->putAwayCooldownTimer == 0) && (this->heldItemAction >= PLAYER_IA_SWORD_MASTER)) {
             Player_UseItem(play, this, ITEM_NONE);
         } else {
-            this->stateFlags2 ^= PLAYER_STATE2_20;
+            this->stateFlags2 ^= PLAYER_STATE2_NAVI_ACTIVE;
         }
     }
 
-    return 0;
+    return false;
 }
```

Comment:
Technically speaking, rolling is the only actual action that this "ActionChange" function concerns. The other things are "extra" and not handled by the main action system. So I could see an argument for naming the function after rolling only. The function comment can explain the "extra" functionality that is also controls. Would like to hear other thoughts.

Reply:
I actually think this is probably for the better and will change it, but still would like to hear opinions